### PR TITLE
Sc 28993/instrument go sttp api

### PR DESCRIPTION
**Pull request by observabuddy**
Problem: We don't have much insight into this API. Libraries that bundle their own loggers are noisy.  

How this addresses the problem: We add basic logger support to the API through the Subscriber struct. This PR uses a `slog.Logger` but doesn't do anything particularly structured with the logs. These changes could be submitted upstream without too much trouble.

It adds Prometheus metrics as well that are registered with the default registry. Those changes are not something we can upstream the way I've done them.

### CODE REVIEW

File: sttp/Subscriber.go [hunk 1]
```diff
@@ -676,6 +687,7 @@ func (sb *Subscriber) handleProcessingComplete(message string) {
 	sb.beginCallbackSync()
 
 	if sb.historicalReadCompleteReceiver != nil {
+		sb.traceCallback(sb.historicalReadCompleteReceiver, "calling historicalReadCompleteReceiver")
 		sb.historicalReadCompleteReceiver()
```
Comment (youngale-pingthings) [inline sttp/Subscriber.go]: Is this going to call this callback twice?

**Review by youngale-pingthings — DISMISSED** [review]

**Comment by pixelherodev** [comment]
No further comments beyond the bug Alex pointed out; looks good otherwise! :)

File: sttp/Subscriber.go (hunk 1 above)
Comment (observabuddy) [reply sttp/Subscriber.go]: Good catch.

**Review by ftrvxmtrx — APPROVED** [review]
lgtm (looks godzilla to me)

**Pull request merged**